src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -3225,12 +3230,14 @@ namespace NServiceBus.Transport
     {
         public TransportReceiveInfrastructure(System.Func<NServiceBus.Transport.IPushMessages> messagePumpFactory, System.Func<NServiceBus.Transport.ICreateQueues> queueCreatorFactory, System.Func<System.Threading.Tasks.Task<NServiceBus.Transport.StartupCheckResult>> preStartupCheck) { }
         public System.Func<NServiceBus.Transport.IPushMessages> MessagePumpFactory { get; }
+        public System.Func<System.Threading.Tasks.Task<NServiceBus.Transport.StartupCheckResult>> PreStartupCheck { get; }
```

Comment:
To be honest I don't like the startup checks. I feel we went over board with thinking ahead and I'm on the fence of exposing those